control-plane/csi-driver/controller/src/pvwatcher.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::Result;
+use futures::TryStreamExt;
+use k8s_openapi::api::core::v1::PersistentVolume;
+
+use kube::{
+    api::{Api, ListParams},
+    runtime::{watcher, WatchStreamExt},
+    Client,
+};
+use tracing::{debug, error, info};
+use uuid::Uuid;
+
+use crate::IoEngineApiClient;
+
+const MAYASTOR_PLUGIN: &str = "io.openebs.csi-mayastor";
+
+pub struct PvGarbageCollector {}
+
+impl PvGarbageCollector {
+    pub async fn start_watcher() -> Result<(), watcher::Error> {
+        info!("Starting PV Garbage Collector");
+        debug!("Handling if any missed events");
+        tokio::spawn(async {
+            handle_missed_events().await;
+        });
+        let client = Client::try_default().await.unwrap();
+        let pv_handle = Api::<PersistentVolume>::all(client.clone());
+        watcher(pv_handle, ListParams::default())
```

Comment:
can we do any filtering here, either with the handle or listParams to watch only our volumes?

Reply:
Filtering can be done in ListParams . For Persistent Volume object we cant do any filtering. I did try it.